step-definitions/Hotels/HotelsSD2.js

```diff
@@ -0,0 +1,177 @@
+            await hpage2.selectLanguage('Español (Estados Unidos)');
+            break;
+        case('English United States'):
+            await hpage2.selectLanguage('English (United States)');
+            break;
+    }
+})
+When(/^I click (save|Guardar) button$/, async function(fieldname){
+    switch(fieldname) {
+        case('save'):
+            await hpage2.clickSave();
+            break;
+        case('Guardar'):
+            await hpage2.clickGuardar();
+            break;
+    }
+})
+When(/^I verify (Español|English) is displayed$/, async function(fieldname){
+    switch(fieldname) {
+        case('Español'):
+            expect(await hpage2.verifyEspanolDisplayed(), 'Espanol not displayed').to.be.true;
+            break;
+        case('English'):
+            expect(await hpage2.verifyEnglishDisplayed(), 'English not displayed').to.be.true;
+    }
+})
+// TC-23
+When(/^I select Jan 10, 2023 as Check-in$/, async function(){
+    await hpage2.selectCheckInDate('January', '2023', '10');
+})
+When(/^I select Jan 23, 2023 as Check-out$/, async function(){
+    await hpage2.selectCheckOutDate('January', '2023', '23');
+})
+When(/^I click on 5 star from star-rating$/, async function(){
+    await srchpage.clickFiveStarRating();
+})
+When(/^I select Price from sort-by dropdown$/, async function(){
+    await srchpage.selectFromSortByDropdown('Price');
+})
+Then(/^I verify all hotels in search results are star-rated as selected in above step$/, async function(){
+    expect(await srchpage.verifyAllHotelsAreSameRating(), 'All hotels are not same rating').to.be.true;
+})
+Then(/^I verify all hotels are listed in increasing order price$/, async function(){
+    expect(await srchpage.verifyPriceLessToMore(), 'Price is not in order').to.be.true;
+})
+//TC-29
+When(/^I click on List your property$/, async function(){
+    await hpage2.clickListYourProp();
+})
+Then(/^I verify “What would you like to list” is displayed$/, async function(){
+    expect(await lyppage.verifyWhatWouldYouLikeisDisplayed(), 'Title not displayed').to.be.true;
+})
+Then(/^I verify “Lodging“ and “Private residence“ options are displayed$/, async function(){
+    expect(await lyppage.verifyLodgingAndPrivateAreDisplayed(), '“Lodging“ and “Private residence“ options are not displayed').to.be.true;
+})
+When(/^I click on “Private residence“$/, async function(){
+    await lyppage.clickPrivate();
+})
+Then(/^I verify ”Step 1 of 3” is displayed$/, async function(){
+    expect(await lyppage.verifyStepOneofThreeDisplayed(), '”Step 1 of 3” is not displayed').to.be.true;
+})
+Then(/^I verify “See how much you could earn!” is displayed$/, async function(){
+    expect(await lyppage.verifyHowMuchEarnDisaplayed(),'“See how much you could earn!” is not displayed').to.be.true;
+})
+When(/^I enter “4“ in bedroom$/, async function(){
+    await lyppage.addBedroom(4);
+})
+When(/^I enter “2.5“ in bathroom$/, async function(){
+    await lyppage.clickAddBathroom(3);
```

Comment:
Instead of passing data like any number/Date/Place/Phone number etc from SD files its better and dynamic to pass the data from feature file steps.
This is not only for this SD but for all step-definitions where you passing data.